vrc-get-gui/app/_main/log/-logs-list-card.tsx

```diff
@@ -27,23 +25,34 @@ export const LogsListCard = memo(function LogsListCard({
 		[logEntry, search, shouldShowLogLevel],
 	);
 
-	const scrollContainerRef = useRef<React.ElementRef<typeof ScrollArea>>(null);
-	const previousLogsLength = useRef(0);
+	const scrollContainerRef = useRef<HTMLDivElement>(null);
 
+	// biome-ignore lint/correctness/useExhaustiveDependencies: should scroll to the bottom whenever the logsShown changes.
 	useEffect(() => {
-		if (autoScroll && scrollContainerRef.current) {
-			if (logsShown.length > previousLogsLength.current) {
-				scrollContainerRef.current.scrollTop =
-					scrollContainerRef.current.scrollHeight;
+		if (!autoScroll) return;
+
+		requestAnimationFrame(() => {
+			if (!scrollContainerRef.current) return;
+
+			const container = scrollContainerRef.current;
+			const isNearBottom =
+				container.scrollHeight -
+					(container.scrollTop + container.clientHeight) <
+				50;
+
+			if (!isNearBottom) {
+				container.scrollTop = container.scrollHeight;
```

Comment:
AS far as I tried and according to react's docs, it looks requestAnimationFrame is not necessary (useEffect will be called after DOM update) so can you try remove wrapping with requestAnimationFrame?